monkey/infection_monkey/control.py

```diff
@@ -42,13 +45,22 @@ def wakeup(parent=None, has_internet_access=None):
         if has_internet_access is None:
             has_internet_access = check_internet_access(WormConfiguration.internet_services)
 
+        user = "Unknown standard user"
+        try:
+            if os.getuid():  # won't throw an exception if it's linux
+                user = os.popen('whoami').read()[:-1]  # get the username
+        except AttributeError:
+            if user_token_is_admin(0):
+                user = "Admin"
```

Comment:
Pretty unclear what the code above does. If you want to check which OS the host is running use a function we already have in `utils.py` or you can add one to check for linux specifically. `user = os.popen('whoami').read()[:-1]` should also be a function, called get_username_monkey_runs_as_in_linux or something similar.